scarlet/morphology.py

```diff
@@ -283,6 +299,39 @@ def get_model(self, *parameters):
         coeffs = self.get_parameter(0, *parameters)
         return starlet_reconstruction(coeffs)
 
+    def update(self):
+        coeffs = self.get_parameter(0)
+        if coeffs.fixed:
+            return
+
+        # shrink the box?
+        image = self.get_model()
+        # image = proxmin.operators.prox_soft(image, 0, thresh=1e-6, type="absolute")
+        bbox = self.bbox.copy()
+        self.shrink_box(image, thresh=1e-8)
+        if bbox != self.bbox:
+            slice, _ = overlapped_slices(bbox, self.bbox)
+            center = tuple(s // 2 for s in self.bbox.shape)
+            constraint = MonotonicMaskConstraint(center, center_radius=1)
```

Comment:
This seems to only work for monotonic constraints. shouldn't it cover the case where constraints are positivity andd L0? Havinng constraint being a class attribute would make it easy.

Reply:
good catch!

I'm not sure what you mean to have the constraint a class attribute. in this case: which class should hold it?

Or put differently, this problem is the reason why I dislike too many options in source classes. I now have to store whether this class actually is `monotonic`.

Reply:
Sorry, I meant making constraint an attribute of the class so that in the update you have `constraint=self.constraint` on line 323 (now)